src/include/sof/common.h

```diff
@@ -9,6 +9,12 @@
 #ifndef __SOF_COMMON_H__
 #define __SOF_COMMON_H__
 
+/*
+ * The most basic alignment macro with no checks, suitable for C, assembly and
+ * for linker scripts
```

Comment:
This comment is misleading: the linker does not use any C macro, the linker has its own ALIGN() function, see `.x.in` files. I explained this in the hard learned comment that this PR is currently dropping :-(

In fact moving this macro outside `#if !defined(LINKER)` is strange because the LINKER doesn't use it.

"suitable for C, assembly" is also misleading because you don't want to use this in C, in C you want to use the better one. The reason this dumber macro exists is for files included in BOTH C _and_ dumber assembly.

On the other hand this PR drops the `#else LINKER` clause that was just added by @lgirdwood . That is good because `common.h` is not used by the linker.

Reply:
@marc-hb I don't think your analysis is quite correct. You're certainly right about linker's own `ALIGN(x)` function, that isn't what I meant. But linker scripts do include normal C headers and use values from them, cf. src/platform/cannonlake/cannonlake.x.in
```
#include <sof/lib/memory.h>
...
PROVIDE(_memmap_vecbase_reset = HP_SRAM_VECBASE_RESET);
```

Reply:
You're right I should have written "does not use ANY C macro". However it uses a small subset and apparently not any sort of ALIGN() macro which is what matters right here. So this comment should not say "suitable for linker scripts".

Reply:
@marc-hb I removed it because I think it's wrong.
```
 * memory.h files require this
 * exact spelling matching the linker function because memory.h values
 * are _also_ copied unprocessed to the .x[.in] linker script
```
I don't think that's correct. I don't see any requirement to use an `ALIGN()` macro in headers to match the `ALIGN()` linker function and isn't this contradicting what you're saying now? Now you're saying that "not any sort of ALIGN() macro" is used in linker scripts from C headers. Or am I misunderstanding anything?